src/modules/Game.class.js

```diff
@@ -1,68 +1,187 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
   constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.initialState =
+      initialState || Array.from({ length: 4 }, () => Array(4).fill(0));
+    this.board = JSON.parse(JSON.stringify(this.initialState));
+    this.status = 'idle';
+    this.score = 0;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  start() {
+    this.status = 'playing';
+    this.addRandomTile();
+    this.addRandomTile();
+  }
+
+  moveDown() {
+    const oldBoard = JSON.stringify(this.board);
+
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    for (let col = 0; col < 4; col++) {
+      let column = this.board.map((row) => row[col]).reverse();
+
+      column = this.combineTiles(column).reverse();
+
+      for (let row = 0; row < 4; row++) {
+        this.board[row][col] = column[row];
+      }
+    }
+
+    if (oldBoard !== JSON.stringify(this.board)) {
+      this.addRandomTile();
+    }
+
+    this.endGame();
+  }
+
+  moveUp() {
+    const oldBoard = JSON.stringify(this.board);
+
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    for (let col = 0; col < 4; col++) {
+      let column = this.board.map((row) => row[col]);
+
+      column = this.combineTiles(column);
+
+      for (let row = 0; row < 4; row++) {
+        this.board[row][col] = column[row];
+      }
+    }
+
+    if (oldBoard !== JSON.stringify(this.board)) {
+      this.addRandomTile();
+    }
+
+    this.endGame();
+  }
```

Comment:
This logic is similar in every method, try to put it in a helper function and use it everywhere.